src/node/setupServer.ts

```diff
@@ -60,6 +67,27 @@ export const setupServer = (...requestHandlers: RequestHandlersList) => {
         const { response } = await getResponse(mockedRequest, currentHandlers)
 
         if (!response) {
+          if (resolvedOptions.onUnhandledRequest === 'warn') {
+            // Produce a developer-friendly warning
+            return console.warn(
+              `A request to ${mockedRequest.url} was detected but not mocked because no request handler matching the URL exists.`,
+            )
+          }
+
+          if (resolvedOptions.onUnhandledRequest === 'error') {
+            // Throw an exception
+
+            // throw new Error(`A request to ${req.url} was detected but not mocked because no request handler matching the URL exists.`)
+            return
+          }
+
+          if (typeof resolvedOptions.onUnhandledRequest === 'function') {
+            resolvedOptions.onUnhandledRequest(mockedRequest)
+            return
+          }
+
+          // resolvedOptions.onUnhandledRequest === 'bypass'
```

Comment:
I would personally avoid duplication of both the error / warn message and the `onUnhandledRequest` logic, by creating a util, like:

```typescript
export const onMissingRequestHandler = (
  request: MockedRequest,
  onUnhandledRequest: OnUnhandledRequest,
) => {
  if (typeof onUnhandledRequest === 'function') {
    onUnhandledRequest(request)
    return
  }

  if (['warn', 'error'].includes(onUnhandledRequest)) {
    const message = `[MSW] A request to ${request.url} was detected but not mocked because no request handler matching the URL exists.`

    if (onUnhandledRequest === 'warn') {
      console.warn(message)
    } else {
      throw new Error(message)
    }
    return
  }

  // if onUnhandledRequest is 'bypass' we have nothing to do.
}
```

I've added `[MSW]` tag at the beginning of the message and refactored a bit the logic.

Reply:
Awesome!